pandas/io/json/normalize.py

```diff
@@ -233,31 +258,11 @@ def _recursive_extract(data, path, seen_meta, level=0):
 
                 _recursive_extract(obj[path[0]], path[1:],
                                    seen_meta, level=level + 1)
-        else:
+        elif isinstance(data, list):
```

Comment:
We only iterate on `data` if it is the list, otherwise we pull the field from the dictionary directly.
The old code always iterates on `data`, which raised the error if `data` is a `dict`.